### ISSUE TITLE
(0.9.0.dev.rc-005) test_create_portfolios fails (either in creation or listing)

### ISSUE THREAD

**Issue by peterkaminski** [GovReady/govready-q#751]
`python3 manage.py test` hits a `list index out of range` in this code:

```
    def test_create_portfolios(self):
        # Create a new account
        self.browser.get(self.url("/"))
        self.click_element('#tab-register')
        self._fill_in_signup_form("test+account@q.govready.com", "portfolio_user")
        self.click_element("#signup-button")

        # Go to portfolio page
        self.browser.get(self.url("/portfolios"))

        # Navigate to portfolio created on signup
        self.click_element_with_link_text("portfolio-user")
```

I'm not sure if this is a failure in creating portfolios, listing portfolios, or a bug in the test.  (It's odd that in one place in the test there is `portfolio_user` (with an underscore), and in another place, there is `portfolio-user` (with a hyphen)).

Test setup and run:

```
git checkout 0.9.0.dev.rc-005
git pull
source env/bin/activate
pip install -r requirements.txt
rm local/db.sqlite3
python3 manage.py migrate
python3 manage.py load_modules
 python3 manage.py first_run
./fetch-vendor-resources.sh
python3 manage.py test
```

Output (note that #750, regarding `test_settings_page` also appears):

```
Creating test database for alias 'default'...
System check identified no issues (0 silenced).
..................login ip=127.0.0.1 username=me result=user:2
.login ip=127.0.0.1 username=me result=user:4
.Elogin ip=127.0.0.1 username=me result=user:12
.login ip=127.0.0.1 username=me result=user:16
..login ip=127.0.0.1 username=me result=user:24
login ip=127.0.0.1 username=me2 result=user:25
login ip=127.0.0.1 username=me3 result=user:26
.login ip=127.0.0.1 username=me result=fail
login ip=127.0.0.1 username=notme result=fail
login ip=127.0.0.1 username=me result=user:28
login ip=127.0.0.1 username=me result=user:28
.login ip=127.0.0.1 username=me result=user:32
.login ip=127.0.0.1 username=me result=user:36
....login ip=127.0.0.1 username=me2 result=user:50
Elogin ip=127.0.0.1 username=me result=user:53
.login ip=127.0.0.1 username=me result=user:57
.login ip=127.0.0.1 username=me result=user:61
.login ip=127.0.0.1 username=me result=user:65
.login ip=127.0.0.1 username=me result=user:69
.
======================================================================
ERROR: test_create_portfolios (siteapp.tests.GeneralTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/kaminski/src/govready/govready-q/siteapp/tests.py", line 596, in test_create_portfolios
    self.click_element_with_link_text("portfolio-user")
  File "/Users/kaminski/src/govready/govready-q/siteapp/tests.py", line 114, in click_element_with_link_text
    elem[0].click()
IndexError: list index out of range

======================================================================
ERROR: test_settings_page (siteapp.tests.OrganizationSettingsTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/kaminski/src/govready/govready-q/siteapp/tests.py", line 1078, in test_settings_page
    response = self.client.get('/settings')
  File "/Users/kaminski/src/govready/govready-q/env/lib/python3.7/site-packages/django/test/client.py", line 535, in get
    response = super().get(path, data=data, secure=secure, **extra)
  File "/Users/kaminski/src/govready/govready-q/env/lib/python3.7/site-packages/django/test/client.py", line 347, in get
    **extra,
  File "/Users/kaminski/src/govready/govready-q/env/lib/python3.7/site-packages/django/test/client.py", line 422, in generic
    return self.request(**r)
  File "/Users/kaminski/src/govready/govready-q/env/lib/python3.7/site-packages/django/test/client.py", line 503, in request
    raise exc_value
  File "/Users/kaminski/src/govready/govready-q/env/lib/python3.7/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/Users/kaminski/src/govready/govready-q/env/lib/python3.7/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/Users/kaminski/src/govready/govready-q/env/lib/python3.7/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/kaminski/src/govready/govready-q/env/lib/python3.7/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/Users/kaminski/src/govready/govready-q/siteapp/views.py", line 1486, in organization_settings
    organization = Organization.objects.get(id=1)
  File "/Users/kaminski/src/govready/govready-q/env/lib/python3.7/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/Users/kaminski/src/govready/govready-q/env/lib/python3.7/site-packages/django/db/models/query.py", line 408, in get
    self.model._meta.object_name
siteapp.models.Organization.DoesNotExist: Organization matching query does not exist.

----------------------------------------------------------------------
Ran 37 tests in 326.981s

FAILED (errors=2)
Destroying test database for alias 'default'...
```

**Comment by gregelin** [comment]
Manually testing the path described within the test `test_create_portfolios (siteapp.tests.GeneralTests)` involves creating a new user account. What is actually failing is creating a new user account. The new user account is not created properly at registration, displaying a debug page. Then "/Users/kaminski/src/govready/govready-q/siteapp/tests.py", line 596 fails because the link to click to create portfolio cannot be found.

```
test_create_portfolios
    self.click_element_with_link_text("portfolio-user")
```

The actual error thrown during account creation is: “You have multiple authentication backends configured and therefore must provide the backend argument or set the backend attribute on the user.”

When I fix that by changing a line in `views_landing.py` from
`login(request, user)` to `login(request, user, 'guardian.backends.ObjectPermissionBackend')`
I then receive a new error of `‘AnonymousUser’ object has no attribute ‘_meta’`.

And that error message I am yet to decipher.

**Comment by gregelin** [comment]
After adding following debug code to `views_landing.py`

```python
                    # Create account.
                    user = signup_form.save(request)
                    print("** debugging new user creation #1")
                    print("user is {}".format(user))
                    # Log them in.
                    from django.contrib.auth import authenticate, login
                    user = authenticate(user_object=user)
                    print("** debugging new user creation #2")
                    print("user is {}".format(user))
                    login(request, user, 'guardian.backends.ObjectPermissionBackend')
```

the error thrown on account creation is trying to login a user that was not authenticated.

```
** debugging new user creation #1
user is foo2
** debugging new user creation #2
user is None
Internal Server Error: /
Traceback (most recent call last):
  File "/codedata/code/govready-q-0.9.0-module-nav/venv/lib/python3.7/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/codedata/code/govready-q-0.9.0-module-nav/venv/lib/python3.7/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/codedata/code/govready-q-0.9.0-module-nav/venv/lib/python3.7/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/codedata/code/govready-q-0.9.0-module-nav/siteapp/views.py", line 35, in homepage
    return homepage(request)
  File "/codedata/code/govready-q-0.9.0-module-nav/siteapp/views_landing.py", line 55, in homepage
    login(request, user, 'guardian.backends.ObjectPermissionBackend')
  File "/codedata/code/govready-q-0.9.0-module-nav/venv/lib/python3.7/site-packages/django/contrib/auth/__init__.py", line 126, in login
    request.session[SESSION_KEY] = user._meta.pk.value_to_string(user)
  File "/codedata/code/govready-q-0.9.0-module-nav/venv/lib/python3.7/site-packages/django/utils/functional.py", line 257, in inner
    return func(self._wrapped, *args)
AttributeError: 'AnonymousUser' object has no attribute '_meta'
```

**Comment by gregelin** [comment]
The problem is that `user` object returned from `signup_form.save(request)` has an encrypted password. So we need to get submitted password from the signup form.

Here is the corrected code, with the first `user` replaced with `new_user`:

```
                    # Create account.
                    new_user = signup_form.save(request)
                    # Log them in.
                    from django.contrib.auth import authenticate, login
                    user = authenticate(request, username=signup_form.cleaned_data['username'], password=signup_form.cleaned_data['password1'])
                    if user is not None:
                        login(request, user, 'django.contrib.auth.backends.ModelBackend')
                    else:
                        print("[ERROR] new_user '{}' did not authenticate after during account creation.".format(new_user.username))
                        messages.error(request, "[ERROR] new_user '{}' did not authenticate during account creation. Account not created. Report error to System Administrator. {}".format(new_user.username, vars(new_user)))
                        return HttpResponseRedirect("/")
```

**Comment by gregelin** [comment]
Fixed in commit 83907df7e2a19df1af338f8c2135992d68d39346
